CMakeLists.txt

```diff
@@ -503,35 +565,85 @@ macro(commom_lib_settings lib write)
 
     if(KTX_FEATURE_VK_UPLOAD)
         target_sources(
-            ${lib}
+            ${target}
         PRIVATE
             include/ktxvulkan.h
             lib/vk_funcs.c
             lib/vk_funcs.h
             lib/vkloader.c
         )
         target_include_directories(
-            ${lib}
+            ${target}
         PRIVATE
             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/lib/dfdutils>
             $<INSTALL_INTERFACE:lib/dfdutils>
         )
 
-        get_target_property( KTX_PUBLIC_HEADER ${lib} PUBLIC_HEADER )
+        get_target_property( KTX_PUBLIC_HEADER ${target} PUBLIC_HEADER )
         list(APPEND KTX_PUBLIC_HEADER ${CMAKE_CURRENT_SOURCE_DIR}/include/ktxvulkan.h)
-        set_target_properties(${lib} PROPERTIES
+        set_target_properties(${target} PROPERTIES
             PUBLIC_HEADER "${KTX_PUBLIC_HEADER}"
         )
     else()
-        target_compile_definitions( ${lib} PRIVATE KTX_OMIT_VULKAN=1 )
+        target_compile_definitions( ${target} PRIVATE KTX_OMIT_VULKAN=1 )
     endif()
 
-endmacro(commom_lib_settings)
+    # Adding write capability to target ktx
+    if(${enable_write})
+        target_sources(
+            ${target}
+        PRIVATE
+            lib/basis_encode.cpp
+            lib/astc_encode.cpp
+            ${BASISU_ENCODER_C_SRC}
+            ${BASISU_ENCODER_CXX_SRC}
+            lib/writer1.c
+            lib/writer2.c
+        )
+    endif()
 
-set(KTX_BUILD_DIR "${CMAKE_BINARY_DIR}")
+    # Setup ktx target
+    if(${enable_write})
```

Comment:
This if and preceding comment seem unnecessary.

Reply:
I kept the original structure as much as I could when I organized this file. I will merge the two ifs.